ui/src/ducks/app/solutions.js

```diff
@@ -130,10 +137,10 @@ export function* createEnvironment(action) {
   yield call(fetchEnvironments);
 }
 
-export function* prepareEnvironment(action) {
+export function* prepareEnvironment(action: {payload: {envName: string, solName: string, solVersion: string }}): Generator<any, void, any> {
   const { envName, solName, solVersion } = action.payload;
 
-  const existingEnv = yield select((state) => state.app.solutions.environments);
+  const existingEnv: SolutionsApi.Environments = yield select((state) => state.app.solutions.environments);
```

Comment:
I think that's a better approach to type effect returns (especially for `call`s) than trying to type the coroutine itself. May be worth generalizing to other sagas.

Reply:
I'm a bit surprised though, don't we have a type mismatch? The annotation L.49 gives:

```javascript
  environments: SolutionsApi.Environment[],
```

But the type definition `SolutionsApi.Environments` describes a map (`{ [name: string]: Environment }`), not a list.